programs/checkout/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    associated_token::AssociatedToken,
+    token::{Mint, Token, TokenAccount},
+};
+
+use solana_program;
+use solana_program::{account_info::AccountInfo, program::invoke, system_instruction};
+
+use exhibition::program::Exhibition;
+use exhibition::{self, Exhibit};
+
+pub mod state;
+use state::heap::Heap;
+use state::linked_list::LinkedList;
+
+declare_id!("8b7yjj2P5fHV9NCyNXJut1pDM1J1D9oRKzqUGW1ycTWk");
+
+#[program]
+pub mod checkout {
+
+    use solana_program::native_token::LAMPORTS_PER_SOL;
+
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        msg!("In initializer");
+
+        let list = LinkedList::initialize();
```

Comment:
This allocates on the stack. You could potentially run out of memory this way if you want to increase the DLL size